internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -248,6 +252,19 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 		}
 	}
 
+	// Loop over all MachineSets and cleanup managed fields.
+	// We do this so that MachineSets that were created/patched before the controller adopted SSA can also work with SSA.
+	// Otherwise, fields would be co-owned by our "old" "manager" and "capi-machinedeployment" and then we would not be able
+	// to e.g. drop labels and annotations.
```

Comment:
wouldn't this be taken care by ForceOwnership?

Reply:
ForceOwnership only takes sole ownership if both field managers disagree over what the field value should be.

If they both are setting the same value they will end up with co-ownership